app/controllers/AdditionalMetadataNavigationController.scala

```diff
@@ -1,252 +1,35 @@
-                                     edges: List[PaginatedFiles.Edges],
-                                     token: BearerAccessToken): Future[List[NodesToDisplay]] = {
-    val nodes = edges.map(edge => {
-      val edgeNode = edge.node
-      val edgeNodeId = edgeNode.fileId
-      val isFolder = edgeNode.fileType.contains("Folder")
-
-      for {
-        descendantFilesSelected <- if (isFolder) { getDescendantFilesSelected(consignmentId, edgeNodeId, token) } else { Future(0) }
-        isSelected <- selected(consignmentId, edgeNodeId, isFolder, token)
-      } yield {
-        NodesToDisplay(
-          edgeNodeId.toString,
-          edgeNode.fileName.getOrElse(""),
-          isSelected,
-          isFolder,
-          descendantFilesSelected
-        )}
+  def getAllFiles(consignmentId: UUID, metadataType: String): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    consignmentService.getAllConsignmentFiles(consignmentId, request.token.bearerAccessToken).map(allFiles => {
+      Ok(views.html.standard.additionalMetadataNavigation(consignmentId, request.token.name, allFiles, metadataType))
     })
-    Future.sequence(nodes)
-  }
-
-  private def selected(consignmentId: UUID, nodeId: UUID, isFolder: Boolean, token: BearerAccessToken): Future[Boolean] = {
-    val folderDescendantsCache = cacheApi.map[List[UUID]](s"${consignmentId}_folders")
-    val selectedFiles: RedisSet[UUID, SynchronousResult] = cacheApi.set[UUID](s"${consignmentId}_selectedFiles")
-
-    for {
-      ids <- if (isFolder) {
-        folderDescendantsCache.getDescendantFileIds(consignmentId, nodeId, token)
-      } else { Future(List(nodeId)) }
-    } yield selectedFiles.containsAll(ids)
   }
 
-  private def getDescendantFilesSelected(consignmentId: UUID,
-                                folderId: UUID,
-                                token: BearerAccessToken): Future[Int] = {
-    val selectedFiles: RedisSet[UUID, SynchronousResult] = cacheApi.set[UUID](s"${consignmentId}_selectedFiles")
-    val partSelectedCache: RedisSet[UUID, SynchronousResult] = cacheApi.set[UUID](s"${consignmentId}_partSelected")
-    val folderDescendantsCache = cacheApi.map[List[UUID]](s"${consignmentId}_folders")
-
-    for {
-      ids <- folderDescendantsCache.getDescendantFileIds(consignmentId, folderId, token)
-      _ = if (selectedFiles.containsSome(ids)) partSelectedCache.add(folderId)
-      descendantFilesSelected = selectedFiles.toSet.intersect(ids.toSet).size
-    } yield descendantFilesSelected
+  def submitFiles(consignmentId: UUID, metadataType: String): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val fileIds = request.body.asFormUrlEncoded.getOrElse(Map()).keys.toList
+      .filter(_ != "csrfToken")
+      .map(UUID.fromString)
+    if(metadataType == "closure") {
+      Future(Redirect(routes.AddClosureMetadataController.addClosureMetadata(consignmentId, fileIds)))
```

Comment:
This should be redirecting to the closure status page: `routes.AdditionalMetadataClosureStatusController.getClosureStatusPage`

Reply:
I've changed it and updated the tests.